modules/tinymce/src/plugins/directionality/main/ts/core/Direction.ts

```diff
@@ -5,28 +5,43 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import Editor from 'tinymce/core/api/Editor';
-import Tools from 'tinymce/core/api/util/Tools';
+import { Arr } from '@ephox/katamari';
+import { Traverse, Attribute, SugarElement } from '@ephox/sugar';
 
-const setDir = (editor: Editor, dir: string) => {
-  const dom = editor.dom;
-  let curDir;
-  const blocks = editor.selection.getSelectedBlocks();
+import Editor from 'tinymce/core/api/Editor';
 
-  if (blocks.length) {
-    curDir = dom.getAttrib(blocks[0], 'dir');
+type Dir = 'rtl' | 'ltl';
 
-    Tools.each(blocks, (block) => {
-      // Add dir to block if the parent block doesn't already have that dir
-      if (!dom.getParent(block.parentNode, '*[dir="' + dir + '"]', dom.getRoot())) {
-        dom.setAttrib(block, 'dir', curDir !== dir ? dir : null);
+const setDir = (editor: Editor, dir: Dir) => {
+  const selectedBlocks = editor.selection.getSelectedBlocks();
+  Arr.each(selectedBlocks, (block) => {
+    const sugarBlock = SugarElement.fromDom(block);
+    const blockParent = Traverse.parent(sugarBlock);
```

Comment:
```suggestion
    const blockParent = Traverse.parent(SugarElement.fromDom(block));
```
Just a style thing here, as we don't need to use `sugarBlock` again we can just inline this.

Reply:
Hmm `sugarBlock` is being used in `setAttrDir` (~line 28) and `Attribute.remove` (else).

Reply:
Ahh, I missed that somehow sorry. Ignore this then.